python/lsst/ts/tunablelaser/hardware.py

```diff
@@ -436,12 +412,59 @@ async def update_register(self):
         None
 
         """
-        await self.power_register.read_register_value()
+        await self.power_register.send_command()
 
     def __repr__(self):
         return f"11PMKu:\n {self.power_register}"
 
 
+class MidiOPG:
+    """Implement the MidiOPG module.
+
+    Parameters
+    ----------
+    component : `Laser`
+        The laser component.
+
+    Attributes
+    ----------
+    name : `str`
+    id : `int`
+    component : `Laser`
+    wavelength_register : `AsciiRegister`
+    """
+
+    def __init__(self, component) -> None:
+        self.name = "MidiOPG"
+        self.id = 31
+        self.component = component
+        self.wavelength_register = AsciiRegister(
+            component=self.component,
+            module_name=self.name,
+            module_id=self.id,
+            register_name="WaveLength",
+            read_only=False,
+            accepted_values=range(1, 2600),
+        )
+
+    async def change_wavelength(self, value):
+        """Change wavelength.
+
+        Parameters
+        ----------
+        value : `float`
+            The wavelength value.
+        """
+        await self.wavelength_register.send_command(value)
+
+    async def update_register(self):
+        """Update all registers."""
+        await self.wavelength_register.send_command()
+
+    def __repr__(self):
+        return f"{self.name}:\n {self.wavelength_register}\n"
+
+
```

Comment:
Please provide a more descriptive class doc for the MaxiOPG and MLDCO48 classes.